fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -590,6 +590,27 @@ public boolean isUnary() {
         }
     }
 
+    @Nullable
+    public static Type invertComparisonType(@Nonnull final Comparisons.Type type) {
+        if (type.isUnary()) {
+            return null;
```

Comment:
`IS_NULL` and `NOT_NULL` are their respective inverses. There doesn't seem to be anything about how this is used that restricts to binary comparisons.

Reply:
Does not need to block this, though.